### PR TITLE
[FEAT]: common table component added.

### PR DESCRIPTION
**Pull request by vaideek-patel**

### CODE REVIEW

File: Client/src/components/pages/Dashboard/AdminDashboard/AdminCategories/index.jsx [hunk 1]
```diff
@@ -68,7 +64,15 @@ const AdminCategories = () => {
             <div className="flex justify-end mb-4">
                 <Link to="/admin-createCategories" className="inline-block px-4 py-2 bg-green-500 text-white rounded hover:bg-green-600 mr-20">+ ADD CATEGORY</Link>
             </div>
-            <Table data={categories} handleUpdate={handleUpdate} handleProductDelete={handleProductDelete} type="category" />
+            {/* <Table data={categories} handleUpdate={handleUpdate} handleProductDelete={handleProductDelete} type="category" /> */}
+            <Table data={categories}
+                handleUpdate={handleUpdate}
+                handleDelete={handleDelete}
+                headers={[
+                    { key: 'id', label: ' ID' },
```
Comment (jay-n-prajapati) [inline Client/src/components/pages/Dashboard/AdminDashboard/AdminCategories/index.jsx]: use variable to store this array and pass it as a prop value in next iteration refactor it

File: Client/src/components/pages/Dashboard/AdminDashboard/AdminUsers/index.jsx [hunk 2]
```diff
@@ -77,39 +82,22 @@ function Index() {
...
-                                            <Link to={`/admin-update/${d.id}`} className="inline-block px-2 py-1 bg-gray-500 text-white rounded hover:bg-gray-600 ml-2">Edit</Link>
-                                            <button className="inline-block px-2 py-1 bg-red-500 text-white rounded hover:bg-red-600 ml-2" onClick={() => handleDelete(d.id)}>Delete</button>
-                                        </td>
-                                    </tr>
-                                ))}
-                            </tbody>
-                        </table>
+
+                        <Table data={slicedData} headers={[
+                            { key: 'id', label: 'ID' },
```
Comment (jay-n-prajapati) [inline Client/src/components/pages/Dashboard/AdminDashboard/AdminUsers/index.jsx]: same as above

File: Client/src/components/pages/Dashboard/AdminDashboard/index.jsx [hunk 3]
```diff
@@ -121,7 +121,19 @@ const Index = () => {
         <Link to="/admin-create-products" className="inline-block px-4 py-2 bg-green-500 text-white rounded hover:bg-green-600 mr-20">+ ADD PRODUCT</Link>
       </div>
 
-      <Table data={paginateRecords} type="product" handleUpdate={handleUpdate} handleProductDelete={handleProductDelete} />
+      {/* <Table data={paginateRecords} type="product" handleUpdate={handleUpdate} handleProductDelete={handleProductDelete} /> */}
+      <Table
+        data={paginateRecords}
+        headers={[
+          { key: 'id', label: 'Product ID' },
```
Comment (jay-n-prajapati) [inline Client/src/components/pages/Dashboard/AdminDashboard/index.jsx]: same as above refactor it

**Review by jay-n-prajapati — CHANGES_REQUESTED** [review]

File: Client/src/components/pages/Dashboard/AdminDashboard/index.jsx (hunk 3 above)
Comment (vaideek-patel) [reply Client/src/components/pages/Dashboard/AdminDashboard/index.jsx]: Okay, Noted, Applying the changes and Re-pushing the code.

File: Client/src/components/pages/Dashboard/AdminDashboard/AdminUsers/index.jsx (hunk 2 above)
Comment (vaideek-patel) [reply Client/src/components/pages/Dashboard/AdminDashboard/AdminUsers/index.jsx]: Okay, Noted, Applying the changes and Re-pushing the code.

File: Client/src/components/pages/Dashboard/AdminDashboard/AdminCategories/index.jsx (hunk 1 above)
Comment (vaideek-patel) [reply Client/src/components/pages/Dashboard/AdminDashboard/AdminCategories/index.jsx]: Okay, Noted, Applying the changes and Re-pushing the code.

**Pull request merged**